app/main/views/suppliers.py

```diff
@@ -411,23 +415,24 @@ def update_supplier_framework_details(supplier_id, framework_slug):
 
     uniform_now = datetime.utcnow()
 
+    agreement_details = interest_record.agreement_details.copy() if interest_record.agreement_details else {}
+
     if 'onFramework' in update_json:
         interest_record.on_framework = update_json['onFramework']
     if 'agreementReturned' in update_json:
         if update_json["agreementReturned"] is False:
             interest_record.agreement_returned_at = None
+            if agreement_details:
+                agreement_details.pop('frameworkAgreementVersion', None)
```

Comment:
If we're clearing `agreementReturned` we probably want to scratch out `agreement_details` altogether (ie set to NULL), not just remove the `frameworkAgreementVersion`?